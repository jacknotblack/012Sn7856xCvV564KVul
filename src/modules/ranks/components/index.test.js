import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Ranks from "./index";

describe("render", () => {
  let wrapper;
  describe("is webview", () => {
    const props = {
      selectedRankType: "sunshine",
      selectedRankPeriod: "weekly",
      onRankTypeClick: jest.fn,
      onRankPeriodClick: jest.fn,
      visibleRanks: [
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 }
      ],
      isWebview: true
    };
    beforeEach(() => {
      wrapper = shallow(<Ranks {...props} />);
    });
    it("renders without crashing", () => {
      expect(wrapper.length).toEqual(1);
      expect(wrapper.find("TestHeader").length).toEqual(0);
      expect(wrapper.find("RankType").length).toEqual(1);
      expect(wrapper.find("RankPeriod").length).toEqual(1);
      expect(wrapper.find(".rank-container").length).toEqual(1);
      expect(wrapper.find("RankItemTop").length).toEqual(3);
      expect(wrapper.find("RankItem").length).toEqual(
        props.visibleRanks.length - 3
      );
    });

    it("should match snapshot", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  describe("non webview", () => {
    const props = {
      selectedRankType: "sunshine",
      selectedRankPeriod: "weekly",
      onRankTypeClick: jest.fn,
      onRankPeriodClick: jest.fn,
      visibleRanks: [
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 },
        { name: "myname", stars: 123 }
      ],
      isWebview: false
    };

    beforeEach(() => {
      wrapper = shallow(<Ranks {...props} />);
    });
    it("renders without crashing", () => {
      expect(wrapper.length).toEqual(1);
      expect(wrapper.find("TestHeader").length).toEqual(1);
      expect(wrapper.find("RankType").length).toEqual(1);
      expect(wrapper.find("RankPeriod").length).toEqual(1);
      expect(wrapper.find(".rank-container").length).toEqual(1);
      expect(wrapper.find("RankItemTop").length).toEqual(3);
      expect(wrapper.find("RankItem").length).toEqual(
        props.visibleRanks.length - 3
      );
    });

    it("should match snapshot", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
