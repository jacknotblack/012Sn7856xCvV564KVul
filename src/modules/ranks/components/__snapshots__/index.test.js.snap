// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render is webview should match snapshot 1`] = `
<div>
  <RankType
    onRankTypeClick={[Function]}
    ranks={
      Array [
        "gastby",
        "diva",
        "sunshine",
      ]
    }
    selectedRankType="sunshine"
  />
  <RankPeriod
    onRankPeriodClick={[Function]}
    periods={
      Array [
        "daily",
        "weekly",
        "monthly",
      ]
    }
    selectedRankPeriod="weekly"
  />
  <div
    className="rank-container"
  >
    <RankItemTop
      bg="bg_first_gastby_ranking.svg"
      key="rankitem123"
      name="myname"
      rankNumber={1}
      stars={123}
    />
    <RankItemTop
      bg="bg_second_ranking.svg"
      key="rankitem123"
      name="myname"
      rankNumber={2}
      stars={123}
    />
    <RankItemTop
      bg="bg_third_ranking.svg"
      key="rankitem123"
      name="myname"
      rankNumber={3}
      stars={123}
    />
    <RankItem
      key="rankitem123"
      name="myname"
      rankNumber={4}
      stars={123}
    />
    <RankItem
      key="rankitem123"
      name="myname"
      rankNumber={5}
      stars={123}
    />
    <RankItem
      key="rankitem123"
      name="myname"
      rankNumber={6}
      stars={123}
    />
    <RankItem
      key="rankitem123"
      name="myname"
      rankNumber={7}
      stars={123}
    />
    <RankItem
      key="rankitem123"
      name="myname"
      rankNumber={8}
      stars={123}
    />
    <RankItem
      key="rankitem123"
      name="myname"
      rankNumber={9}
      stars={123}
    />
    <RankItem
      key="rankitem123"
      name="myname"
      rankNumber={10}
      stars={123}
    />
  </div>
</div>
`;

exports[`render non webview should match snapshot 1`] = `
<TestHeader>
  <div>
    <RankType
      onRankTypeClick={[Function]}
      ranks={
        Array [
          "gastby",
          "diva",
          "sunshine",
        ]
      }
      selectedRankType="sunshine"
    />
    <RankPeriod
      onRankPeriodClick={[Function]}
      periods={
        Array [
          "daily",
          "weekly",
          "monthly",
        ]
      }
      selectedRankPeriod="weekly"
    />
    <div
      className="rank-container"
    >
      <RankItemTop
        bg="bg_first_gastby_ranking.svg"
        key="rankitem123"
        name="myname"
        rankNumber={1}
        stars={123}
      />
      <RankItemTop
        bg="bg_second_ranking.svg"
        key="rankitem123"
        name="myname"
        rankNumber={2}
        stars={123}
      />
      <RankItemTop
        bg="bg_third_ranking.svg"
        key="rankitem123"
        name="myname"
        rankNumber={3}
        stars={123}
      />
      <RankItem
        key="rankitem123"
        name="myname"
        rankNumber={4}
        stars={123}
      />
      <RankItem
        key="rankitem123"
        name="myname"
        rankNumber={5}
        stars={123}
      />
      <RankItem
        key="rankitem123"
        name="myname"
        rankNumber={6}
        stars={123}
      />
      <RankItem
        key="rankitem123"
        name="myname"
        rankNumber={7}
        stars={123}
      />
      <RankItem
        key="rankitem123"
        name="myname"
        rankNumber={8}
        stars={123}
      />
      <RankItem
        key="rankitem123"
        name="myname"
        rankNumber={9}
        stars={123}
      />
      <RankItem
        key="rankitem123"
        name="myname"
        rankNumber={10}
        stars={123}
      />
    </div>
  </div>
</TestHeader>
`;
